gui/default/syncthing/core/syncthingController.js

```diff
@@ -2778,9 +2778,21 @@ angular.module('syncthing.core')
 
             $scope.restoreVersions.tree.filterNodes(function (node) {
                 if (node.folder) return false;
-                if ($scope.restoreVersions.filters.text && node.key.indexOf($scope.restoreVersions.filters.text) < 0) {
-                    return false;
+
+                if ($scope.restoreVersions.filters.text) {
+                    // Use case-insensitive filter.
+                    var filterText = $scope.restoreVersions.filters.text.toLowerCase();
+                    var versionPath = node.key.toLowerCase();
+                    // Allow backslashes on Windows as filter path separators.
+                    if ($scope.version.os == 'windows') {
+                        filterText = filterText.replace(/\//g, '\\');
+                        versionPath = versionPath.replace(/\//g, '\\');
```

Comment:
If I understand this correctly, you're replacing every slash with a backslash on Windows?  The regex is kinda hard to read because of all the slashes involved for different purposes.

Why not go the other way round?  Doesn't Syncthing use forward-slashes for relative paths within a folder?  Converting backslashes to slashes would make more sense to me, as it reduces differences between OSes.

Reply:
Wouldn't this interfere with using backslashes for escape on other OSs, e.g. `my\ file`? As far as I'm aware, you can also have backslashes legitimately as a part of the filename too (but not in Windows).

Reply:
Afaik we use native path separators here, so `/` everywhere except windows where it's `\\`. However that would make this change a noop on the `versionPath` side of things, so I guess you checked and it's using forward slashes here @tomasz1986 ? I had a quick look at the code and didn't see anything that suggest that we use forward slashes everywhere, but I might have easily missed the place where a translation happens.

Reply:
Yeah, the tree uses forward slashes which I think come from here:

https://github.com/syncthing/syncthing/blob/f42f041f5329f08117624d79fe25ed5e695f5e85/gui/default/syncthing/app.js#L176-L214

Reply:
@acolomb I've changed the conversion as you asked and improved the comment to make it clear what is going on!

Reply:
> Yeah, the tree uses forward slashes which I think come from here:

Then there's no need/point in changing `versionPath`, right?

Reply:
Except if someone expects to paste a (partial) Windows path into the search box and have it match hierarchical paths with their forward slashes.  Thus I'd keep it.